cmake/sdp.cmake

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+function(sdp_assembly_generate)
+  set(hrt_dir ${PROJECT_SOURCE_DIR}/src/hrt)
+  set(hrt_src ${hrt_dir}/hrt.c)
+  set(hrt_msg "Generating ASM file for Hard Real Time files.")
+
+  # The external static library that we are linking with does not know
+  # how to build for this platform so we export all the flags used in
+  # this zephyr build to the external build system.
+  zephyr_get_compile_options_for_lang(C hrt_opt)
+  zephyr_get_compile_definitions_for_lang(C hrt_def)
+  zephyr_get_include_directories_for_lang(C hrt_inc)
+  zephyr_get_system_include_directories_for_lang(C hrt_sys)
+  # Replace "-I" with "-isystem" to treat all Zephyr headers as system headers
+  # that do not trigger -Werror.
+  string(REPLACE "-I" "-isystem" hrt_inc "${hrt_inc}")
+
+  zephyr_include_directories(include)
+
+  add_custom_target(asm_gen
```

Comment:
please specify `BYPRODUCTS`.

Also, I can see this create a custom `asm_gen` target, but afaict there is no use of this target / its byproducts in the build system, and I see no documentation describing the intention or usage of this target.

How / when is this target intended to be invoked ?

Reply:
This is just a part of a larger work related to generating assembler files for the Hard-Real Time part of SDP. This PR only aims to add a custom target in the form of `asm_gen`, which will be used in subsequent PRs.

Reply:
please let the original commenter decide if the comment has been resolved.

This is not done:
> please specify BYPRODUCTS.


also, regarding this:
> This is just a part of a larger work related to generating assembler files for the Hard-Real Time part of SDP. 

That might be, but without corresponding code I cannot review this function in depth, namely I cannot verify incremental builds or check that byproducts are correctly removed on a `ninja clean`.

If I approve this PR and later discover issues with the code in your next PR which uses the function , then that just results in a reply to me that, `but this function was approved in PR#17638` (talking by experience).

So I need at least the follow up / usage of this code to be able to verify the full behavior of this code.

Reply:
@tejlmand here is my draft PR that uses `asm_gen`: #18018. The intention is to run `asm_check` either in west build after the CMake configuration and before the building starts, or directly in CMake at the beginning of a build. We want to run `asm_check` in every SDP build, so that if someone changes the hrt/.c file, they know that the ASM file should be updated (the .c file will not be included in the build, it only serves as a source to generate the ASM file. The ASM file will be included in the build instead). We do not want to update the ASM file by default because it may change solely because a different compiler was used - which is what we are trying to avoid here. 

Please, note that my PR is still a work in progress, thus calling `asm_check` is not added yet.

Also, if anything is done incorrectly in this PR, it is most likely also done incorrectly in mine, as I was basing my work on it.